### PR TITLE
Unified snapshots configuration for datanode

### PR DESCRIPTION
**Pull request by todvora**
This PR refactors and fixes opensearch snapshots configuration in datanode. It recognizes if either s3 or local filesystem repos are configured and only then adds the `search` node role and configures `node.search.cache.size`. 

This limits disk space usage in datanode setups without searchable snapshots enabled and avoids unexpected watermark warnings and errors.

Additionally there is a validation of usable disk space and cache size configuration. If the cache would consume all the usable space, an exception is thrown. If the cache size exceeds 80% of usable space, a warning will be logged.

## How Has This Been Tested?
Added unit tests, existing set of integration tests for valid startups.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: data-node/src/main/java/org/graylog/datanode/Configuration.java [hunk 1]
```diff
@@ -254,7 +254,10 @@ public class Configuration {
     @Parameter(value = "metrics_policy")
     private String metricsPolicy = "gl-datanode-metrics-ism";
 
-    @Documentation(value = "Cache size for searchable snaphots")
+    /**
+     * @see <a href="https://opensearch.org/docs/latest/tuning-your-cluster/availability-and-recovery/snapshots/searchable_snapshot/#configuring-a-node-to-use-searchable-snapshots}">Searchable snapshots</a>
+     */
+    @Documentation(value = "Cache size for searchable snaphots. ")
     @Parameter(value = "node_search_cache_size")
```
Comment (moesterheld) [inline data-node/src/main/java/org/graylog/datanode/Configuration.java:261]: would it make sense to use a `NamedBindingOverride` getter here to configure it to a different value if no s3 repo is configured?

**Review by moesterheld — APPROVED** [review]
LGTM. However, I would think we could brainstorm about whether we can set the cache size dynamically

File: data-node/src/main/java/org/graylog/datanode/Configuration.java (hunk 1 above)
Comment (todvora) [reply data-node/src/main/java/org/graylog/datanode/Configuration.java:261]: There are two types of searchable snaphsot repos - s3 and shared filesystem. Looking at it now, we could automatically  minimize the cache size based on either s3 configuration presence OR `path_repo` configured. If users want to use snapshots, they need to give us one or the other. So we could take it as an indicator that the cache is needed :thinking:

Comment (moesterheld) [reply data-node/src/main/java/org/graylog/datanode/Configuration.java:261]: yes, that sounds good. i will change the pr back to requested changes

Comment (todvora) [reply data-node/src/main/java/org/graylog/datanode/Configuration.java:261]: refactored, now we check both s3 and path_repo config and based on that decide if we add the search role and configure the cache.

**Review by moesterheld — CHANGES_REQUESTED** [review]
Tested and lgtm in general.
However, I am missing the pass-through for the `path.repo` setting. Or am I missing something?

**Review by moesterheld — APPROVED** [review]
never mind, I was not looking in the right place.

**Pull request merged**